pkgs/development/compilers/rust/make-rust-platform.nix

```diff
@@ -17,59 +18,64 @@
   ...
 }:
 
-rec {
-  fetchCargoTarball = buildPackages.callPackage ../../../build-support/rust/fetch-cargo-tarball {
-    git = buildPackages.gitMinimal;
-    inherit cargo;
-  };
+(makeScopeWithSplicing' {
+  otherSplices = generateSplicesForMkScope "rustPlatform";
+  f =
+    self:
+    let
+      inherit (self) callPackage;
+    in
+    {
+      fetchCargoTarball = buildPackages.callPackage ../../../build-support/rust/fetch-cargo-tarball {
+        git = buildPackages.gitMinimal;
+        inherit cargo;
+      };
 
-  fetchCargoVendor = buildPackages.callPackage ../../../build-support/rust/fetch-cargo-vendor.nix {
-    inherit cargo;
-  };
+      fetchCargoVendor = buildPackages.callPackage ../../../build-support/rust/fetch-cargo-vendor.nix {
+        inherit cargo;
+      };
 
-  buildRustPackage = callPackage ../../../build-support/rust/build-rust-package {
-    inherit
-      stdenv
-      cargoBuildHook
-      cargoCheckHook
-      cargoInstallHook
-      cargoNextestHook
-      cargoSetupHook
-      fetchCargoTarball
-      fetchCargoVendor
-      importCargoLock
```

Comment:
This was the root cause of the inconsistency. Since these hooks were consumed directly in their non-spliced form here, they did not get their platform shifts applied inside buildRustPackage, which made the hooks behave differently here than in other packages where they were consumed via `rustPlatform.xxx`.